api_app/models.py

```diff
@@ -181,3 +188,60 @@ def delete_file(sender, instance: Job, **kwargs):
     if instance.file:
         if instance.file:
             instance.file.delete()
+
+
+class CustomConfig(models.Model):
+    class Type(models.IntegerChoices):
+        ANALYZER = 1, "Analyzer"
+        CONNECTOR = 2, "Connector"
+
+    type = models.PositiveSmallIntegerField(choices=Type.choices)
+    attribute = models.CharField(max_length=128, blank=False)
+    value = models.CharField(max_length=128, blank=False)
+    organization = models.ForeignKey(
+        Organization, on_delete=models.CASCADE, blank=True, null=True
+    )
+    owner = models.ForeignKey(
+        settings.AUTH_USER_MODEL, on_delete=models.CASCADE, blank=False, null=False
+    )
+    name = models.CharField(max_length=128, blank=False)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def get_as_dict(cls, user, entity_type, name=None) -> dict:
+        """
+        Returns custom config as dict
+        """
+        custom_configs = QuerySet(cls)
+
+        # Since, user-level custom configs should override organization-level configs,
+        # we need to get the organization-level configs, if any, first.
+        try:
+            membership = Membership.objects.get(user=user)
+            custom_configs = custom_configs | cls.objects.filter(
```

Comment:
cool usage of bitwise operator for dictionaries. But we can make it even more cool. Try `|=` to avoid to repeat `custom_configs`. Please use that also in the other parts where you leveraged with operator.